tests/metrics_test.go

```diff
@@ -153,14 +158,95 @@ var _ = Describe("Metrics", func() {
 		)
 	})
 
+	Context("SSP metrics", func() {
+		getDeployment := func(name string, namespace string) *apps.Deployment {
+			deployment := &apps.Deployment{}
+			err := apiClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployment)
+			Expect(err).ToNot(HaveOccurred())
+			return deployment
+		}
+
+		getOperatorPod := func() (k8sv1.Pod, uint16) {
+			pods, port := operatorPodsWithMetricsPort()
+			return pods[0], port
+		}
+
+		updateVersion := func(version string) {
+			deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
+
+			envs := deployment.Spec.Template.Spec.Containers[0].Env
+			var newEnvs []k8sv1.EnvVar
+
+			for _, env := range envs {
+				if env.Name != common.OperatorVersionKey {
+					newEnvs = append(newEnvs, env)
+				}
+			}
+
+			newEnvs = append(newEnvs, k8sv1.EnvVar{Name: common.OperatorVersionKey, Value: version})
+			deployment.Spec.Template.Spec.Containers[0].Env = newEnvs
+
+			err := apiClient.Update(ctx, deployment)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		BeforeEach(func() {
+			updateVersion("devel")
+			waitUntilDeployed()
+		})
+
+		AfterEach(func() {
+			updateVersion("devel")
+		})
+
+		It("should increment total_restored_common_templates when not in upgrade", func() {
+			sspPod, port := getOperatorPod()
+
+			By("getting initial restored count")
+			restoredCount := intMetricValue("total_restored_common_templates", port, &sspPod)
+
+			By("creating test template")
+			testTemplate := createTestTemplate()
+
+			By("getting restored count after creating test template")
+			expectRestoreAfterUpdate(&testTemplate)
+			newRestoredCount := intMetricValue("total_restored_common_templates", port, &sspPod)
+			Expect(newRestoredCount).To(Equal(restoredCount + 1))
+		})
+
+		It("should not increment total_restored_common_templates during upgrades", func() {
```

Comment:
The test is missing `[test_id:]`. Once this PR is good to merge, we can ask QE for an ID. In the meantime, can you add `[test_id:TODO]` here, so we know it is missing?

Reply:
added